x-pack/plugins/observability_solution/asset_manager/server/routes/enablement/uninstall_builtin_definitions.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RequestHandlerContext } from '@kbn/core/server';
+import { getFakeKibanaRequest } from '@kbn/security-plugin/server/authentication/api_keys/fake_kibana_request';
+import { readEntityDiscoveryAPIKey, checkIfEntityDiscoveryAPIKeyIsValid } from '../../lib/auth';
+import { SetupRouteOptions } from '../types';
+import { ENTITY_INTERNAL_API_PREFIX } from '../../../common/constants_entities';
+import { ERROR_API_KEY_NOT_FOUND, ERROR_API_KEY_NOT_VALID } from '../../../common/errors';
+import { uninstallBuiltInEntityDefinitions } from '../../lib/entities/uninstall_entity_definition';
+
+export function uninstallBuiltinEntityDefinitionsRoute<T extends RequestHandlerContext>({
+  router,
+  server,
+  logger,
+}: SetupRouteOptions<T>) {
+  router.post(
+    {
+      path: `${ENTITY_INTERNAL_API_PREFIX}/managed/stop`,
+      validate: false,
+    },
+    async (context, req, res) => {
+      try {
+        logger.debug('reading entity discovery API key from saved object');
+        const apiKey = await readEntityDiscoveryAPIKey(server);
```

Comment:
can we put the 'read and check' logic into some helper function? it's used twice (or three times?) in this PR, and probably will be used again in the future